src/jvm/main/org/jetbrains/kotlinx/lincheck/runner/FixedActiveThreadsExecutor.kt

```diff
@@ -21,8 +21,8 @@ import java.util.concurrent.locks.*
 /**
  * This executor maintains the specified number of threads and is used by
  * [ParallelThreadsRunner] to execute [ExecutionScenario]-s. The main feature
- * is that this executor keeps the re-using threads "hot" (active) as long as
- * possible, so that they should not be parked and unparked between invocations.
+ * is that this executor keeps the re-using threads "hot" (active) as long as possible,
+ * so that they should not be parked and unparked between invocations.
  */
 internal class FixedActiveThreadsExecutor(private val nThreads: Int, runnerHash: Int) : Closeable {
```

Comment:
I remember that we had an issue leading to a hang. Do you provide a fix in this PR?

Reply:
We have this bug in our issue tracker (https://github.com/JetBrains/lincheck/issues/194) and tests in Kotlin Coroutines now hang as well

Reply:
> I remember that we had an issue leading to a hang. Do you provide a fix in this PR?

> We have this bug in our issue tracker (https://github.com/JetBrains/lincheck/issues/194) and tests in Kotlin Coroutines now hang as well

No, sorry, I do not know what is a cause of this bug, and I haven't tried to address it directly in this PR. 

This PR handles the situations when actors hung in init/post parts, but I suppose it is a different issue.

Reply:
The bug has already been fixed and will be merged in develop shortly